stats/opentelemetry/client_metrics.go

```diff
@@ -163,15 +210,29 @@ func (h *clientStatsHandler) TagRPC(ctx context.Context, info *stats.RPCTagInfo)
 		}
 		ctx = istats.SetLabels(ctx, labels)
 	}
-	ai := &attemptInfo{ // populates information about RPC start.
+	ai := &attemptInfo{
 		startTime: time.Now(),
 		xdsLabels: labels.TelemetryLabels,
 		method:    info.FullMethodName,
 	}
-	ri := &rpcInfo{
-		ai: ai,
+	if !isTracingDisabled(h.options.TraceOptions) {
+		callSpan := trace.SpanFromContext(ctx)
+		if info.NameResolutionDelay {
+			callSpan.AddEvent("Delayed name resolution complete")
+		}
+		var newAI *attemptInfo
+		ctx, newAI = h.traceTagRPC(trace.ContextWithSpan(ctx, callSpan), info)
+		// Update the ai with values from updated attempt info.
+		newAI.startTime = ai.startTime
```

Comment:
this is weird. why not update `ai` with values needed from `newAI`? Also, it can be just tinfo since it will contain information about trace only and then the trace related field should be set in ai. We should eventually return ai